flags.go

```diff
@@ -142,23 +186,117 @@ func (g *goforit) Enabled(ctx context.Context, name string) (enabled bool) {
 
 	g.flagsMtx.RLock()
 	defer g.flagsMtx.RUnlock()
+	enabled = false
 	if g.flags == nil {
-		enabled = false
+		g.logger.Printf("[goforit] empty flags map")
 		return
 	}
 	lastRefreshTime = g.lastFlagRefreshTime
 	flag := g.flags[name]
 
-	// equality should be strict
-	// because Float64() can return 0
-	if f := g.rand(); f < flag.Rate {
+	// if flag is inactive, always return false
+	if !flag.Active {
+		return
+	}
+
+	var mergedProperties = map[string]string{}
+
+	g.tagsMtx.RLock()
+	for k, v := range g.defaultTags {
+		mergedProperties[k] = v
+	}
+	g.tagsMtx.RUnlock() // should i defer this? maybe inside its own scope?
+
+	for k, v := range properties {
+		mergedProperties[k] = v
+	}
+
+	// if there are no rules, but flag is active, always return true
+	if len(flag.Rules) == 0 {
```

Comment:
This is confusing. Shouldn't "continue off the end of the rules" always have the same behavior?